ls1intum/Artemis
# `Exam mode:` Move assessment buttons into exam assessment dashboard

### Checklist
#### General


#### Client
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/). **I moved the existing client tests**


### Motivation and Context
The "Evaluate Quizzes" and "Assess unsubmitted exams" buttons are assessment related but were located in the "Student Exams" page. This was confusing and we decided to change it.

### Description
 I moved the two buttons into their own component and import this component in the assessment dashboard, effectively moving the buttons to the assessment dashboard page. I also added some bootstrap classes to the component to make sure its in the correct location. I then also refactored the existing client tests to cover the new component and also changed existing cypress e2e tests which also test related functionality.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Exam with 1 Quiz Exercise

1. As student make a submission in the exam
2. As instructor navigate to the assessment dashboard
3. Check that the buttons are inactive before the exam has ended
4. Check that the buttons become active after the exam has ended
5. Check that clicking the buttons results in the correct notification being shown
6. Check that after using the button and after the exam results are released the student has received the quiz assessment

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots


**Before:**
![image](https://user-images.githubusercontent.com/56715566/143719256-640afdfa-ff97-42d2-96a7-802a8673afa9.png)

**After:**
![image](https://user-images.githubusercontent.com/56715566/143691688-a26edb48-abaf-4a02-bf73-03bf7a7d9e9a.png)
![image](https://user-images.githubusercontent.com/56715566/143691854-c069e283-301b-4d01-882d-7d8004591284.png)
